src/ApiProblem.php

```diff
@@ -411,7 +407,13 @@ public function asJson(bool $pretty = false)
             $options = JSON_UNESCAPED_SLASHES | JSON_PRETTY_PRINT;
         }
 
-        return json_encode($response, $options);
+        $json = json_encode($response, $options);
+
+        if (false === $json) {
+            throw new \RuntimeException('Unable to generate encoded json');
+        }
```

Comment:
Is this even possible?  I'm assuming not unless compile() is horribly broken.  If it is, it should probably check json_last_error() like the fromJson() method does and throw a more precise exception.  (Also, exception text should end with a period.)

Reply:
With strict types it's better to handle it. I used [phpstan](https://github.com/phpstan/phpstan) for static analyze the code.
I created another exception to handle it. No other exceptions end with a period .